[STACK iview2dcm -> refactor] Make iView to DICOM app

SimonBiggs opened the pull request:
Alright, so I am certainly stepping beyond my expertise here. Anyway... I think this is ready for review.

sjswerdloff commented on lib/pymedphys/_dicom/constants/core.py:
```diff
@@ -37,12 +37,6 @@
 
 from pymedphys._imports import pydicom
 
-# Many thanks to the Medical Connections for offering free
```
Why are credits being deleted?   Was it moved?

SimonBiggs replied:
Credits aren't being deleted. This was actually duplicated within the `constants/uuid.py` file. So, this duplication was deleted and the other file was used instead.

sjswerdloff commented on lib/pymedphys/_experimental/streamlit/apps/iview_to_dicom.py:
```diff
@@ -72,19 +144,60 @@ def _create_portal_image_dicom_dataset(
+            # as this would require appropriately grouping the images
+            # from each field. Certainly doable, and arguably it should
+            # be done, but out of scope for the current task.
+            "PatientID": patient_id,
```
OK, this addresses a comment I made elsewhere.
Very good, happy to see it

SimonBiggs replied:
> OK, this addresses a comment I made elsewhere.

Do you mean this comment:

https://github.com/pymedphys/pymedphys/pull/1387#issuecomment-774913098

Does that mean the `base` branch can be approved and merged?

sjswerdloff replied:
Yes.
I think not providing the UID hierarchy early on in the construction 
is a mistake but it is your mistake to make.

SimonBiggs replied:
Kk... if you feel strongly about this, I can see I have can expand on it appropriately on Thursday?

sjswerdloff commented on lib/pymedphys/_experimental/streamlit/apps/iview_to_dicom.py:
```diff
@@ -72,19 +144,60 @@ def _create_portal_image_dicom_dataset(
+            "Modality": "RTIMAGE",
+            "InstanceCreatorUID": PYMEDPHYS_ROOT_UID,
+            "SOPClassUID": "1.2.840.10008.5.1.4.1.1.481.1",
+            "SOPInstanceUID": pydicom.uid.generate_uid(prefix=f"{PYMEDPHYS_ROOT_UID}."),
```
Study and Series uids also.

SimonBiggs replied:
Does the following address your concerns:

https://github.com/pymedphys/pymedphys/blob/ba2d8dc483b88e1a13d59623847f4038a3455127/lib/pymedphys/_experimental/streamlit/apps/iview_to_dicom.py#L165-L168

The same logic applies for series.

For now, it appears these files do import into pips. I don't want to create study or series UIDs at the moment. In the future I imagine these sorts of things could be passed in as a parameter to a more general function, and then that more general function could be called by a "study" or "series" creator function.

But... that's not anywhere within my scope at the moment.

sjswerdloff commented on lib/pymedphys/_experimental/streamlit/apps/iview_to_dicom.py:
```diff
@@ -72,19 +144,60 @@ def _create_portal_image_dicom_dataset(
+            "PixelData": pixel_array.tobytes(),
             "RadiationMachineSAD": sad,
             "RTImageSID": sid,
+            "PrimaryDosimeterUnit": "",
```
MU should be fine.
If they are KV images, I'll need to look it up, but having something might be better than empty, and MU covers the vast majority of situations and systems

SimonBiggs replied:
For this one, I'd prefer to err on the side of agreeing with what is currently exported from the iView database by the official tool, which, in this case is `""`.

sjswerdloff replied:
> For this one, I'd prefer to err on the side of agreeing with what is currently exported from the iView database by the official tool, which, in this case is `""`.

I will submit a defect to the vendor

SimonBiggs replied:
🙂

sjswerdloff reviewed (changes requested):
Are you decompressing the jpeg?  Compatibility will be much better with uncompressed data.

Requested change is just to use MU for primary dosimeter unit

SimonBiggs commented:
> Are you decompressing the jpeg? Compatibility will be much better with uncompressed data.

Yup, that's what `pymedphys._losslessjpeg` is doing.

SimonBiggs commented:
@sjswerdloff when I believe I have addressed all concerns raised within the stack, did you want me to press the "request re-review" button to flag it for your attention?

![image](https://user-images.githubusercontent.com/6559099/107300626-01be7b80-6ace-11eb-8105-a401983c10af.png)

sjswerdloff commented:
> @sjswerdloff when I believe I have addressed all concerns raised within the stack, did you want me to press the "request re-review" button to flag it for your attention?
> 
> ![image](https://user-images.githubusercontent.com/6559099/107300626-01be7b80-6ace-11eb-8105-a401983c10af.png)

Yes

SimonBiggs commented:
> Yes

kk 🙂, will do. Just did it over at:

https://github.com/pymedphys/pymedphys/pull/1412
